projects/plugins/jetpack/modules/calypsoify/class-jetpack-calypsoify.php

```diff
@@ -2,210 +2,145 @@
 /**
  * This is Calypso skin of the wp-admin interface that is conditionally triggered via the ?calypsoify=1 param.
  *
+ * @deprecated $$next-version$$ Use Automattic\Jetpack\Calypsoify instead.
+ *
  * @package automattic/jetpack
  */
 
 use Automattic\Jetpack\Calypsoify\Jetpack_Calypsoify as Calypsoify;
-use Automattic\Jetpack\Status;
 
 /**
  * Class Jetpack_Calypsoify
+ *
+ * @deprecated $$next-version$$
  */
 class Jetpack_Calypsoify {
 
 	/**
 	 * Singleton instance of `Jetpack_Calypsoify`.
 	 *
+	 * @deprecated $$next-version$$
+	 *
 	 * @var object
 	 */
 	public static $instance = false;
 
 	/**
 	 * Is Calypsoify enabled, based on any value of `calypsoify` user meta.
 	 *
+	 * @deprecated $$next-version$$
+	 *
 	 * @var bool
 	 */
 	public $is_calypsoify_enabled = false;
 
-	/**
-	 * Jetpack_Calypsoify constructor.
-	 */
-	private function __construct() {
-		add_action( 'admin_init', array( $this, 'setup' ), 4 );
-	}
-
 	/**
 	 * Singleton.
 	 *
+	 * @deprecated $$next-version$$
+	 *
 	 * @return Jetpack_Calypsoify
 	 */
 	public static function get_instance() {
-		if ( ! self::$instance ) {
-			self::$instance = new self();
-		}
-
-		return self::$instance;
+		_deprecated_function( __METHOD__, 'jetpack-$$next-version$$', 'Automattic\\Jetpack\\Calypsoify\\Jetpack_Calypsoify::get_instance' );
+		return Automattic\Jetpack\Calypsoify\Jetpack_Calypsoify::get_instance();
 	}
 
 	/**
 	 * Setup function that is loaded on the `wp_loaded` hook via the constructor.
+	 *
+	 * @deprecated $$next-version$$
 	 */
 	public function setup() {
-		$this->is_calypsoify_enabled = isset( $_GET['calypsoify'] ) && 1 === (int) $_GET['calypsoify'] && $this->is_page_gutenberg(); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
-
-		$this->check_meta();
-
-		if ( $this->is_calypsoify_enabled ) {
-			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_for_gutenberg' ), 100 );
-		}
+		_deprecated_function( __METHOD__, 'jetpack-$$next-version$$', 'Automattic\\Jetpack\\Calypsoify\\Jetpack_Calypsoify::setup' );
+		return ( new Automattic\Jetpack\Calypsoify\Jetpack_Calypsoify() )->setup();
```

Comment:
To do this right in a normal case, what we could do is something like this:
```php
class Whatever {
    /** @var \Automattic\Jetpack\Whatever */
    private $wrapped;

    public function __construct() {
        _deprecated_function( __METHOD__, 'jetpack-$$next-version$$', 'Automattic\\Jetpack\\Whatever::__construct' );
        $this->wrapped = new \Automattic\Jetpack\Whatever();
    }

    public function something() {
        _deprecated_function( __METHOD__, 'jetpack-$$next-version$$', 'Automattic\\Jetpack\\Whatever::something' );
        return $this->wrapped->something();
    }
}
```
But as mentioned above, the constructors are private so even getting an instance to call this `setup()` on would be kind of hard.

I suppose another way to do it for this situation might be like
```php
class Jetpack_Calypsoify extends \Automattic\Jetpack\Calypsoify\Jetpack_Calypsoify {

    private function __construct() {
    }

    public static function get_instance() {
        _deprecated_function( __METHOD__, 'jetpack-$$next-version$$', 'Automattic\\Jetpack\\Calypsoify\\Jetpack_Calypsoify::get_instance' );
        return parent::get_instance();
    }

    public function setup() {
        _deprecated_function( __METHOD__, 'jetpack-$$next-version$$', 'Automattic\\Jetpack\\Calypsoify\\Jetpack_Calypsoify::setup' );
        parent::setup();
    }

    // and so on
}
```
That would also transparently handle the `\Jetpack_Calypsoify::$instance` variable, since it would access the parent class's copy.

Reply:
> I suppose another way to do it for this situation might be like

This option appears to work well, thanks for the suggestion.